app/Http/Controllers/Web/GroupsController.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Rogue\Http\Controllers\Web;
+
+use Rogue\Models\Group;
+use Illuminate\Http\Request;
+use Rogue\Http\Controllers\Controller;
+
+class GroupsController extends Controller
+{
+    /**
+     * Create a controller instance.
+     */
+    public function __construct()
+    {
+        $this->middleware('auth');
+        $this->middleware('role:admin,staff');
+
+        $this->rules = [
+            'name' => 'required|unique:groups,group_type_id',
```

Comment:
I poked around the docs, but I'm still kinda shaky on how this unique rule works. Is the `group_type_id` saying the `name` is only unique within children of the same group type?

Reply:
I'm just going to remove this `unique` because I couldn't get it working, and add a TODO to revisit. We want a unique index on `group_type_id` and `name` for groups, this attempts to validate before we try to save the group -- and it didn't do the trick.